File: android/src/main/java/com/getcapacitor/community/classes/results/ConnectedPeersResult.java
package com.getcapacitor.community.classes.results;

import com.getcapacitor.JSObject;
import com.getcapacitor.community.interfaces.Result;

import java.util.List;
import java.util.UUID;

public class ConnectedPeersResult implements Result {

    private final List<UUID> peers;

    public ConnectedPeersResult(List<UUID> peers) {
        this.peers = peers;
    }

    @Override
    public JSObject toJSObject() {
        JSObject result = new JSObject();

        if (peers != null) result.put("peers", peers.toArray());

        return result;
    }
}

File: ios/Sources/BridgefyPlugin/Classes/Results/ConnectedPeersResult.swift
import Foundation
import Capacitor

@objc public class ConnectedPeersResult: NSObject, Result {
    let connectedPeers: [UUID]?

    init(_ connectedPeers: [UUID]?) {
        self.connectedPeers = connectedPeers
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()

        if let connectedPeers = connectedPeers {
            var jsPeers = JSArray()

            jsPeers.reserveCapacity(connectedPeers.count)

            for peer in connectedPeers {
                jsPeers.append(peer.uuidString)
            }

            result["peers"] = jsPeers
        }

        return result as AnyObject
    }
}
